File: Java/0201-0250/217. Contains Duplicate.java
// https://leetcode.com/problems/contains-duplicate/description/

class Solution {
    public boolean containsDuplicate(int[] nums) {
        //u can sort the array and find the duplicates or use hashset to reduce time complexity
        HashSet<Integer> storage = new HashSet<>();

        //traverse through the given array
        for(int index=0; index<nums.length; index++){
            if(storage.contains(nums[index])){
                return true; //duplicates found
            }
            //add the element to the hashSet
            storage.add(nums[index]);
        }

        return false; //no duplicates found
    }
}

File: Python/0201-0250/217. Contains Duplicate.py
# https://leetcode.com/problems/contains-duplicate/description/

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # sort the array and find the duplicate or use Set for reducing time complexity
        storage = {""}

        for num in nums:
            # check if set contains the num
            if num in storage:
                return True # duplicate found
            storage.add(num)
        
        return False # No duplcicates found
